src/Castle.Core/DynamicProxy/Generators/MethodFinder.cs

```diff
@@ -78,17 +78,14 @@ private static MethodInfo[] MakeFilteredCopy(MethodInfo[] methodsInCache, Bindin
 
 		private static object RemoveDuplicates(MethodInfo[] infos)
```

Comment:
(Unrelated to this change, and I haven't looked at the call sites... but I wonder why this method's declared return type is `object`, instead of `MethodInfo[]`? Do call sites have to cast the return value back to `MethodInfo[]`?)

Reply:
They get stored into a field `Dictionary<Type, object> cachedMethodInfosByType`, so yes this field should be updated too since there is only one call site of this private method.